application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/TripPatternForDate.java

```diff
@@ -224,6 +223,20 @@ public TripPatternForDate newWithFilteredTripTimes(Predicate<TripTimes> filter)
     return new TripPatternForDate(tripPattern, filteredTripTimes, filteredFrequencies, serviceDate);
   }
 
+  /**
+   * Asserts that the running period is valid and throws an {@link IllegalArgumentException} if it
+   * is not.
+   * This is the last line of defense against invalid real-time updates.
+   */
+  public void assertValidRunningPeriod() throws IllegalArgumentException {
+    assertValidRunningPeriod(
+      startOfRunningPeriod,
+      endOfRunningPeriod,
+      tripTimes[0],
+      tripTimes[tripTimes.length - 1]
+    );
+  }
```

Comment:
Perhaps just get rid of the arguments and include the code from the private function into this public one. Also, I think the throw/catch pattern here probably doesn't make sense anymore. I think you can just return a boolean.

Reply:
Although, I don't have a strong opinion on this. Now that I think more about this, the try/catch can make sense.

Reply:
Lets discuss in the meeting.